packages/SwingSet/docs/vat-environment.md

```diff
@@ -0,0 +1,119 @@
+# The Vat Runtime JS Environment
```

Comment:
Just a question - we've had significant problems in the past with documentation being put in markdown here in the agoric-sdk repo and never getting to the documentation site, where the developers will be looking for it. Can we have a rule against putting documentation files in agoric-sdk?

Reply:
Hm, good point. On the other hand, having it next to the code means it's possible to make a PR that changes both the implementation and the code (and reviewers should complain if the API is modified without also updating the docs, and that provides a good place to take a critical look at the doc changes and see if we're really ok with changing our API surface). I added this because I didn't find (ok maybe didn't look very hard for) a description of what vat code can expect to use, and this PR is potentially changing that. I couldn't tell if there was some e.g. global name that vats/contracts will expect to have in scope, which I was maybe breaking.

I'm not sure how to reconcile those two goals.

What if we have the docs here, next to the code, but changes to them should be reviewed by @tyg, who can use the information to decode how/if to update the out-of-repo docs site?

Reply:
I like having the primary docs next to the code. The Zoe contracts have descriptions in the Documentation repo, but they haven't been kept up-to-date. That's why I added a bunch of documentation inside the contract sources, and hoped that the details would be transported to the Documentation site as well. We should develop a process that makes it possible for TYG to do something about it.